app/views/links/history.html.erb

```diff
@@ -0,0 +1,51 @@
+<div class="left">
+    <%= link_to "Back to link", url_for(action: :show) %>
+</div>
+
+<h2>Previous Wallpapers</h2>
+<p>Below is a list of the last <%= pluralize(@past_links.length, "wallpaper")%> set by others for this link. Click any wallpaper to go to see the full size image.</p>
+<div class="link-history--list">
+  <% @past_links.each do |past_link| %>
+    <% unless past_link.post_thumbnail_url.nil? %>
+      <div class="link-history--card" data-controller="past-reaction">
+        <% if past_link.horny? %>
+          😍
+        <% end %>
+        <% if past_link.came? %>
+          💦
+        <% end %>
+        <a href="<%= past_link.post_url %>" class="link-history--image-link" target="_blank">
+          <%= image_tag past_link.post_thumbnail_url %>
+        </a>
+        <div class="link-history--card-footer"> 
+          <% unless past_link.set_by.nil? %>
+            <%= link_to past_link.set_by.username, user_path(past_link.set_by.username), target: '_top' %>
+          <% else %>
+            anon
+          <% end %>
+          <%= button_tag class: 'no-underline secondary', data: { action: 'click->past-reaction#addReaction' } do %>
+            <ion-icon role="presentation" name="happy"></ion-icon>
+            <ion-icon role="presentation" name="add"></ion-icon>
+          <% end %>
+        </div>
+        <dialog class="modal link-history--modal" data-past-reaction-target="modal">
```

Comment:
This is kinda awkward, the modal's markup is repeated once for each possible entry in the history. I'm not really sure if there's a better way to do this without shoveling some logic into the JS

Reply:
ok, this garbage has been ripped out in favor of just having the server send the markup of an individual modal when i click. way more sane than this nonsense.